cpp/models/abm/household.cpp

```diff
@@ -22,59 +22,68 @@
 #include "memilio/math/eigen.h"
 #include <string>
 
-namespace mio {
+namespace mio
+{
+namespace abm
+{
 
-namespace{
+namespace
+{
 /**
  * Picks an age from a custom index array with a weight for each age group according to a discrete distribution.
  * @param age_groups A custom index array with the weights.
  * @return The picked age group.
  */
-mio::AbmAgeGroup pick_age_group_from_age_distribution(const mio::CustomIndexArray<int, mio::AbmAgeGroup>& age_groups){
+AgeGroup pick_age_group_from_age_distribution(const CustomIndexArray<int, AgeGroup>& age_groups)
+{
     auto age_group_weights = age_groups.array().cast<double>().eval();
-    size_t age_group = mio::DiscreteDistribution<size_t>::get_instance()(age_group_weights);
-    return (mio::AbmAgeGroup) age_group;
-}
+    size_t age_group       = DiscreteDistribution<size_t>::get_instance()(age_group_weights);
+    return (AgeGroup)age_group;
 }
+} // namespace
```

Comment:
anonymous namespaces solve ODR (one definition rule) linker errors. There must be exactly one definition for each function/class/variable in all compilation units that are linked together into one binary. It is safer to put functions internal to one compilation unit (i.e. not exported by the header) into an anonymous namespace in case a function of the same name is defined in another compilation unit. Otherwise the linker could not distinguish between the two functions and would silently drop one of them